### PR TITLE
Improvements to check-merge-conflicts action

### PR DESCRIPTION
**Pull request by barrettj12**
Add a Go script which parses the git log and determines who might have created merge conflicts. It will determine which files are conflicted, and notify the authors of any commit which touched these files.

Also add the ability to ignore certain commit authors (e.g. jujubot, dependabot).

## Example run

A test run of the new workflow is [here](https://github.com/barrettj12/juju/actions/runs/5407952219/jobs/9826511030).

1. It attempts to merge the current branch (`merge-check-blame`) into the merge target branch (`3.3`). The merge encounters conflicts and is unsuccessful.

2. We then go through the git log, looking at all the files that are conflicted (`--merge` flag), and determine all commits since the last merge that have touched these files.
   ```
   {"sha":"b048634105eba3d66ade23b0a64c69d1097dbd81","author":"Harry Pidcock","email":"harry.pidcock@canonical.com"}
   {"sha":"498f6ff9c0f83704d19b863b8f59269297e6841f","author":"Jack Shaw","email":"jack.shaw@canonical.com"}
   {"sha":"874fd8ed70fb0c0939faa0cd7d1b68d02df27771","author":"Harry Pidcock","email":"harry.pidcock@canonical.com"}
   {"sha":"69548bce394c6f47055e14db663afe35a028ad3e","author":"Harry Pidcock","email":"harry.pidcock@canonical.com"}
   ```

3. We determine the authors of these commits (removing duplicates), map their emails to Mattermost usernames, and generate a nice notification message to be sent to Juju Watercooler.
   > @\hpidcock, @\jack-shaw: your recent changes to `merge-check-blame` have caused merge conflicts. Please merge `merge-check-blame` into `3.3` and resolve the conflicts.

### CODE REVIEW

File: .github/workflows/merge.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 name: Merge
 on:
   push:
-    branches: ['2.9', '3.1', '3.2', '3.3']
+    branches: ['2.9', '3.1', '3.2', '3.3', 'merge-check-blame']
```
Comment (hpidcock) [inline .github/workflows/merge.yml]: Obviously drop this before merging.

File: .github/workflows/merge.yml [hunk 2]
```diff
@@ -13,6 +13,7 @@ jobs:
         3.1: 3.2
         3.2: 3.3
         3.3: main
+        merge-check-blame: 3.3
```
Comment (hpidcock) [inline .github/workflows/merge.yml]: and this

File: scripts/try-merge/execute.go [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+	command string
+	args    []string
+	dir     string
+}
+
+type executeResults struct {
+	runError error
+	exitCode int
+
+	stdout, stderr, combined []byte
```
Comment (hpidcock) [inline scripts/try-merge/execute.go:22]: combined?

**Review by hpidcock — APPROVED** [review]

File: scripts/try-merge/execute.go (hunk 3 above)
Comment (barrettj12) [reply scripts/try-merge/execute.go:22]: unused

**Pull request merged**